lib/src/Base/Func/TensorizedUniVariateFunctionFactory.cxx

```diff
@@ -0,0 +1,151 @@
+
+typedef Collection<NumericalPoint> NumericalPointCollection;
+typedef ProductUniVariateFunctionEvaluationImplementation::UniVariateFunctionCollection UniVariateFunctionCollection;
+
+/* Default constructor */
+TensorizedUniVariateFunctionFactory::TensorizedUniVariateFunctionFactory()
+  : FunctionalBasisImplementation()
+  , coll_()
+  , phi_()
+{
+  // Nothing to do
+}
+
+
+/* Constructor */
+TensorizedUniVariateFunctionFactory::TensorizedUniVariateFunctionFactory(const FunctionFamilyCollection & coll)
+  : FunctionalBasisImplementation()
+  , coll_(coll)
+  , phi_(coll.getSize())
+{
+  // Nothing to do
+}
+
+
+/* Constructor */
+TensorizedUniVariateFunctionFactory::TensorizedUniVariateFunctionFactory(const FunctionFamilyCollection & coll,
+    const EnumerateFunction & phi)
+  : FunctionalBasisImplementation(),
+    coll_(coll),
+    phi_(phi)
+{
+  if (coll.getSize() != phi.getDimension()) throw InvalidArgumentException(HERE) << "Error: the enumerate function must have a dimension equal to the collection size";
+}
+
+
+/* Virtual constructor */
+TensorizedUniVariateFunctionFactory * TensorizedUniVariateFunctionFactory::clone() const
+{
+  return new TensorizedUniVariateFunctionFactory(*this);
+}
+
+
+/* Return the enumerate function that translate unidimensional indices into multidimensional indices */
+EnumerateFunction TensorizedUniVariateFunctionFactory::getEnumerateFunction() const
+{
+  return phi_;
+}
+
+/* set the enumerate function that translate unidimensional indices into multidimensional indices */
+void TensorizedUniVariateFunctionFactory::setEnumerateFunction(const EnumerateFunction & phi) 
+{
+  phi_ = phi;
+}
+
+/* set the function familly collection */
+void TensorizedUniVariateFunctionFactory::setFunctionFamilyCollection(const FunctionFamilyCollection & coll)
+{
+  coll_ = coll;
+  if (coll.getSize() != phi_.getDimension())
+  {
+    setEnumerateFunction(EnumerateFunction(coll.getSize()));
+  }
+}
+
+/* Build the NumericalMathFunction of the given index */
+NumericalMathFunction TensorizedUniVariateFunctionFactory::build(const UnsignedInteger index) const
+{
+  // Compute the multi-indices using the EnumerateFunction
+  Indices indices(phi_(index));
+  const UnsignedInteger size = indices.getSize();
+  // Then build the collection of functions using the collection of factories
+  UniVariateFunctionCollection functions(size);
+  for (UnsignedInteger i = 0; i < size; ++ i)
+  {
+    functions[i] = coll_[i].build(indices[i]);
+  }
+  const Pointer<ProductUniVariateFunctionEvaluationImplementation> p_evaluation(ProductUniVariateFunctionEvaluationImplementation(functions).clone());
+  return NumericalMathFunctionImplementation(p_evaluation,
+         ProductUniVariateFunctionGradientImplementation(p_evaluation).clone(),
+         ProductUniVariateFunctionHessianImplementation(p_evaluation).clone());
+}
+
+
+/* String converter */
+String TensorizedUniVariateFunctionFactory::__repr__() const
```

Comment:
__repr__ is here to give the maximum amount of information about the object, so it should list all the attributes with there exact names and values
__str__ is here for the pretty-print of the object, and it is up to the developer to hide or rename some of the attributes